src/components/Input/Input.scss

```diff
@@ -5,43 +5,126 @@
     @extend %input;
 
     border-color: var(--color-neutral-20);
-
-    &--context {
-        @each $context in ($contexts) {
-            &_#{ $context } {
-                border-color: var(--color-#{$context}-50);
-                &:focus {
-                    border-color: var(--color-#{$context}-50);
-                }
-            }
-        }
-    }
+    border-radius: var(--space-100);
+    font-size: var(--font-size-small);
 
     &::placeholder {
-        color: var(--color-neutral-30);
+        color: var(--color-text-subtlest);
+        line-height: var(--line-height-small);
         opacity: 1;
     }
 
     &--isBlock {
         width: 100%;
     }
 
+    &:hover {
+        background-color: var(--color-background-input-hover);
+        border: 1px solid var(--color-border-input);
+    }
+
+    &:focus {
+        background-color: inherit;
+        border: 1px solid var(--color-info-20);
+    }
+
     &[disabled] {
-        background-color: var(--color-background);
+        background-color: var(--color-background-input-default);
+        color: var(--color-text-disabled);
+        border: 1px solid var(--color-border-disabled);
+
+        &::placeholder {
+            color: var(--color-text-disabled);
+        }
+
+        &:hover,
+        &:focus {
+            background-color: var(--color-background-input-default);
+            border: 1px solid var(--color-border-disabled);
+        }
+    }
+
+    &[readOnly] {
+        background-color: var(--color-background-input-read-only);
+        border: none;
+
+        &:hover,
+        &:focus {
+            background-color: var(--color-background-input-read-only);
+            border: none;
+        }
+    }
+
+    &--isActive {
```

Comment:
While trying to make `:focus-visible` and `:focus` work differently on an input component, I noticed they behaved the same way, not allowing differentiation between keyboard and mouse clicks. This seems to be intentional and is related to the user agent heuristics. I found several discussions on this topic, but this one explains it best: [GitHub issue on WICG/focus-visible](https://github.com/WICG/focus-visible/issues/131). 
It appears that for elements like `button` or `link`, the focus styles can differ based on the interaction method, but for `input` fields, the focus is designed to be consistent regardless of whether it's via mouse, touch, or keyboard.
So I leave it as it is, what else we can do it since it is not feasible via css we could have only one focus style.

Reply:
Nice investigation! 
Maybe we shouldn't have this active state then? Sounds like a non standard behaviour, and I don't see the value in it while it can lead to bugs... do you want to check with Dima? I'm happy to join the chat if that is helpful

Reply:
Yeah, I totally agree, I will talk to him... because it feels weird to implement something what is not even intended to happen. I will talk to him and update the code

Reply:
We agreed to remove isActive state and use only `focused` one, the style for `focused` should be as it was for `active` state